src/oop/static_modifier/hard/q6/MainTest.java

```diff
@@ -0,0 +1,223 @@
+        Object[][] constructorArgs = {
+                {value1, double.class},
+                {value2, int.class}
+        };
+        Object checkupInstance = testClass.createInstance(constructorArgs);
+        _assertEquals(value1, testClass.getFieldValue(checkupInstance, attributeName1), String.format(wrongValueMessage, className, attributeName1));
+        _assertEquals(value2, testClass.getFieldValue(checkupInstance, attributeName2), String.format(wrongValueMessage, className, attributeName2));
+    }
+
+    @Test
+    @Order(5)
+    public void receiveDonationIsDefinedCorrectly() {
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(methodName2, null), String.format(incorrectMethodDefinition, methodName2, className));
+        assertTrue(testClass.hasModifier(methodName2, null, "public"), String.format(incorrectModifierMessage, methodName2, className));
+        assertTrue(testClass.hasReturnType(methodName2, null, Void.TYPE), String.format(incorrectReturnType, methodName2, className));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    @Tag("dependency1")
+    @Order(6)
+    public void farmClassHasCorrectReceiveDonationMethod(double value1, int value2, double value3) throws Throwable {
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(methodName2, null), String.format(incorrectMethodDefinition, methodName2, className));
+        assertTrue(testClass.hasModifier(methodName2, null, "public"), String.format(incorrectModifierMessage, methodName2, className));
+        assertTrue(testClass.hasReturnType(methodName2, null, Void.TYPE), String.format(incorrectReturnType, methodName2, className));
+        Object[][] constructorArgs = {
+                {value1, double.class},
+                {value2, int.class}
+        };
+        Object classInstance = testClass.createInstance(constructorArgs);
+        testClass.setFieldValue(null, value3, staticAttributeName);
+        Clause[] testSentence;
+        if (value1 * value2 <= value3) {
+            testSentence = new Clause[]{
+                    new StringLiteral("Thank you for your donation."),
```

Comment:
make these strings variables at the top of the tests

Reply:
The period in it should also be escaped since periods have a special meaning in regex